e2e/simple_ondatra_test.go

```diff
@@ -0,0 +1,217 @@
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	startReq := &mpb.StartRequest{
+		From: mirror.Port(t, "port1").Name(),
+		To:   mirror.Port(t, "port2").Name(),
+	}
+	if _, err := client.Start(ctx, startReq); err != nil {
+		t.Fatalf("cannot start mirror client, got err: %v", err)
+	}
+}
+
+func stopMirror(t *testing.T, client mpb.MirrorClient) {
+	t.Helper()
+	mirror := ondatra.DUT(t, "mirror")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	stopReq := &mpb.StopRequest{
+		From: mirror.Port(t, "port1").Name(),
+		To:   mirror.Port(t, "port2").Name(),
+	}
+	if _, err := client.Stop(ctx, stopReq); err != nil {
+		t.Fatalf("cannot stop mirror client, got err: %v", err)
+	}
+}
+
+func TestMirror(t *testing.T) {
+	addr := mirrorAddr(t)
+	client, stop := mirrorClient(t, addr)
+	defer stop()
+	startMirror(t, client)
+	time.Sleep(1 * time.Second)
+	stopMirror(t, client)
+}
+
+// TestMPLS is a simple test that creates an MPLS flow between two ATE ports and
+// checks that there is no packet loss. It validates magna's end-to-end MPLS
+// flow accounting.
+func TestMPLS(t *testing.T) {
+	// Start a mirroring session to copy packets.
+	maddr := mirrorAddr(t)
+	client, stop := mirrorClient(t, maddr)
+	defer stop()
+	startMirror(t, client)
+	time.Sleep(1 * time.Second)
+	defer func() { stopMirror(t, client) }()
+
+	otgCfg := pushBaseConfigs(t, ondatra.ATE(t, "ate"))
+
+	otg := ondatra.ATE(t, "ate").OTG()
+	otgCfg.Flows().Clear().Items()
+	mplsFlow := otgCfg.Flows().Add().SetName("MPLS_FLOW")
+	mplsFlow.Metrics().SetEnable(true)
+	mplsFlow.TxRx().Port().SetTxName(ateSrc.Name).SetRxName(ateDst.Name)
+
+	mplsFlow.Rate().SetChoice("pps").SetPps(1)
+
+	// Set up ethernet layer.
+	eth := mplsFlow.Packet().Add().Ethernet()
+	eth.Src().SetChoice("value").SetValue(ateSrc.MAC)
+	eth.Dst().SetChoice("value").SetValue(ateDst.MAC)
+
+	// Set up MPLS layer with destination label.
+	mpls := mplsFlow.Packet().Add().Mpls()
+	mpls.Label().SetChoice("value").SetValue(destinationLabel)
+	mpls.BottomOfStack().SetChoice("value").SetValue(0)
+
+	mplsInner := mplsFlow.Packet().Add().Mpls()
+	mplsInner.Label().SetChoice("value").SetValue(innerLabel)
+	mplsInner.BottomOfStack().SetChoice("value").SetValue(1)
+
+	ip4 := mplsFlow.Packet().Add().Ipv4()
+	ip4.Src().SetChoice("value").SetValue("100.64.1.1")
+	ip4.Dst().SetChoice("value").SetValue("100.64.1.2")
+	ip4.Version().SetChoice("value").SetValue(4)
+
+	otg.PushConfig(t, otgCfg)
+
+	t.Logf("Starting MPLS traffic...")
+	otg.StartTraffic(t)
+	t.Logf("Sleeping for %s...", *sleepTime)
+	time.Sleep(*sleepTime)
+	t.Logf("Stopping MPLS traffic...")
+	otg.StopTraffic(t)
+
+	// Avoid a race with telemetry updates.
+	time.Sleep(1 * time.Second)
+	metrics := gnmi.Get(t, otg, gnmi.OTG().Flow("MPLS_FLOW").State())
+	got, want := metrics.GetCounters().GetInPkts(), metrics.GetCounters().GetOutPkts()
+	lossPackets := want - got
```

Comment:
nit: move inline (on 214)? doesn't look like lossPackets is used elsewhere

Reply:
Done, thanks.